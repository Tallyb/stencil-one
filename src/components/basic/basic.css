.nice {
    background: blue;
    color: honeydew;

}