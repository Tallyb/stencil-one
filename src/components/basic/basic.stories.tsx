import { h } from '@stencil/core';

export default {
  title: 'My Basic',
};

export const Default = () => (<my-basic first="Millie" middle="Bobby" last="Brown"></my-basic>);

