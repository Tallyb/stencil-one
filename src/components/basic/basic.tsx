import { Component, Prop, h, JSX } from '@stencil/core';

@Component({
  tag: 'my-basic',
  styleUrl: 'basic.css',
  shadow: true
})
export class MyBasic {

    @Prop() first: string;

    @Prop() last: string;
  
    render(): JSX.Element {
      return (
        <div>
          <p class="nice">My name is {this.last} {this.first}</p>
          <slot/>
        </div>
            
      );
    }
}