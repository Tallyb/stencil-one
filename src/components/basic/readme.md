# my-basic



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type     | Default |
| -------- | --------- | ----------- | -------- | ------- |
| `first`  | `first`   |             | `string` | `''`    |
| `last`   | `last`    |             | `string` | `''`    |


## Dependencies

### Used by

 - [my-nested](../nested)

### Graph
```mermaid
graph TD;
  my-nested --> my-basic
  style my-basic fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
