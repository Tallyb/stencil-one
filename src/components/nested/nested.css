.nice {
    background: palevioletred;
    color: blueviolet;

}