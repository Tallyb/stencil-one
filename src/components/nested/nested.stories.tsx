/** @jsx h */
/** @jsxRuntime classic */

import {h} from '@stencil/core';

export default {
  title: 'My Nested',
};

export const Default = {
  render: () => (<my-nested first="Millie" last="Brown"></my-nested>)
};

