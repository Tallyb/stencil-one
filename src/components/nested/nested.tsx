import { Component, Prop, h, JSX } from '@stencil/core';

@Component({
  tag: 'my-nested',
  styleUrl: 'nested.css',
  shadow: true
})
export class MyNested {

    @Prop() first: string = '';

    @Prop() last: string = '';
  

    render(): JSX.Element {
      return (
        <div>
          <my-basic first={'Sean'} last={'Connery'}/>
          <p class="very-nice">My name is {this.last} {this.first}</p>
        </div>            
      );
    }
}