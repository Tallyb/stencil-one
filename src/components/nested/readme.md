# my-basic



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type     | Default |
| -------- | --------- | ----------- | -------- | ------- |
| `first`  | `first`   |             | `string` | `''`    |
| `last`   | `last`    |             | `string` | `''`    |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
