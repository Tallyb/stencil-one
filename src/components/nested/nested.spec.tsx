import { newSpecPage } from '@stencil/core/testing';
import {h} from '@stencil/core';
import { MyNested } from './nested';
import { MyBasic } from '../basic/basic';


describe('nested - using template', () => {
  const template = () => (
    <my-nested first={'Julia'} last={'Roberts'}>
    </my-nested>
  );
  const components = [MyNested, MyBasic];

  it('should render with shadow dom', async () => {
    const page = await newSpecPage({
      template,
      components,
    })!;
    // expect(page.root.sh).toEqualHtml('aaa');
    expect(page.root?.shadowRoot).toBeTruthy();
    const el: HTMLElement = page.root?.shadowRoot?.querySelector('.very-nice')!;
    expect(el.innerHTML).toEqual('My name is Roberts Julia');
    expect(page.root?.shadowRoot?.querySelector('my-basic')).toBeTruthy();
    expect(page.root).toMatchSnapshot();
  });
});
