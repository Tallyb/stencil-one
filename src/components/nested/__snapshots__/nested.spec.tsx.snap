// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nested - using template should render with shadow dom 1`] = `
<my-nested>
  <mock:shadow-root>
    <div>
      <my-basic>
        <mock:shadow-root>
          <div>
            <p class="nice">
              My name is Connery Sean
            </p>
            <slot></slot>
          </div>
        </mock:shadow-root>
      </my-basic>
      <p class="very-nice">
        My name is Roberts Julia
      </p>
    </div>
  </mock:shadow-root>
</my-nested>
`;
